querydsl-jpa/src/main/java/com/querydsl/jpa/impl/AbstractJPAQuery.java

```diff
@@ -208,10 +214,20 @@ private Object getSingleResult(Query query) {
     @Override
     public QueryResults<T> fetchResults() {
```

Comment:
Should we deprecate `fetchResults`? It's easy to fetch the count separately if the user wants it.

Reply:
`QueryResults` seems to be a very intentional QueryDSL API. 

I think is mostly an unfortunate naming decision in QueryDSL 4. `list()` was way more recognizable than `fetch()`, so users would less likely unintentionally end up calling `listReslults()` instead of `fetchResults()`.

Its even more problematic in JPA where you also have `fetchJoin` which adds the FETCH join specifier for initializing associations with the query. Having both `fetch()` and `fetchJoin` makes the methods even more ambiguous.

Personally I'd really like to rename `fetch()` to `fetchResults()` and `fetchResults()` to something like `fetchQueryResults()` (or remove it altogether), but its going to be a rather breaking change. Definitely only possible for 5.x and even then probably not received fairly well with users struggling with once again porting their application.

I am not sure what to do at this point, but I would be fine with deprecating `fetchResults()`. At least that gives us a mechanism to direct users to use `fetch()`  instead. However, I am not sure if deprecating `fetchResults` makes sense if we do not intend to remove or rename it eventually. I am not sure how widely it is "actively" used.

Reply:
I think deprecating it now makes sense. Then we'd be able to rename it/remove it in 5.X or 6.X.

Reply:
And then deprecate it on `AbstractJPAQuery` only or `Fetchable`? (the latter would also affect `querydsl-sql` and others, which might support `fetchResults()` just fine)

Reply:
I would deprecate it on `Fetchable`. `fetchResults` might work correctly in other modules, but, in my opinion, the method just creates confusion and unwanted side effects over functionality that could easily be added in the application.

Reply:
I've deprecated it on `AbstractJPAQuery` and added a detailed description in the Javadoc regarding rationale and alternatives.